### PR TITLE
Combining data scripts

### PR DESCRIPTION
**Pull request by pkrshnan**
What was changed:
1. Combined `video_frames.py` and `labelling_frames.py` in one script named `split_video.py`.
2. Added optional compression option to reduce quality to 70% of the original.
3. Created a timestamps folder that is used to label objects in certain time periods of training videos, consisting of CSV files.

Left to do:
1. ~Add data augmentation option to `split_video.py` script, in order to create more randomized training data.~ - this should be part of the training script, not the video frame script.

Should be good for a review.

### CODE REVIEW

File: timestamps/1142.csv [hunk 1]
```diff
@@ -0,0 +1,6 @@
+start,end,title
```
Comment (Waterloo-Aquadrone) [inline timestamps/1142.csv:1]: Given the way you're splitting, does it not read this line as input timestamps (and give an error?)

File: preprocessing/creatingData/split_video.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+# Optional Arguments:
+#   --compress: optionally compress image after saving
+
+import cv2
+import os
+import argparse
+
+from PIL import Image
+
+# Construct the argument parser
```
Comment (Waterloo-Aquadrone) [inline preprocessing/creatingData/split_video.py]: should be in main function

File: preprocessing/creatingData/split_video.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+ap.add_argument("-d", "--directory", required=True,
+    help="path to the directory to store the frames. Will create a new directory \
+    if needed")
+ap.add_argument("-f", "--file", required=True,
+    help="path to the text file containing timestamps for objects in video")
+ap.add_argument("-c", "--compress", action="store_true",
+    help="compress images before saving")
+args = vars(ap.parse_args())
+
+def getFrame(time, framerate):
```
Comment (Waterloo-Aquadrone) [inline preprocessing/creatingData/split_video.py:16]: Can you add docstrings to functions? ie
def getFrame(time, framerate):
   """
   A function to get the frame number from a given time and framerate
   inputs:
   time: the time as a string, in format hh:mm:ss
   framerate: the framerate as a float
   outputs:
   the frame number as an integer
   """
   insert code here

File: timestamps/1142.csv (hunk 1 above)
Comment (pkrshnan) [reply timestamps/1142.csv:1]: Nope, because I run:

```python
header = file.readline()
```
which skips the first line of the file. That exists just in case we want to verify if the header is correct in the future, if for example, our timestamps files get more complicated.

Comment (Waterloo-Aquadrone) [reply timestamps/1142.csv:1]: Neat all good then!

**Review by ghost — APPROVED** [review]
